### ISSUE TITLE
Consider adding a warning for captioned figures that contain a single captioned table

### ISSUE THREAD

**Issue by EricDunsworth** [validator/validator#1208]
**URL of WHATWG spec or PR:**
https://html.spec.whatwg.org/multipage/tables.html#the-caption-element:the-table-element-5

**Excerpt:**
> When a [table](https://html.spec.whatwg.org/multipage/tables.html#the-table-element) element is the only content in a [figure](https://html.spec.whatwg.org/multipage/grouping-content.html#the-figure-element) element other than the [figcaption](https://html.spec.whatwg.org/multipage/grouping-content.html#the-figcaption-element), the [caption](https://html.spec.whatwg.org/multipage/tables.html#the-caption-element) element should be omitted in favor of the [figcaption](https://html.spec.whatwg.org/multipage/grouping-content.html#the-figcaption-element).

**Describe the solution you'd like**
I wanted to propose showing a validation warning if the validator detects the following triggers:
* A ``figure`` element contains a ``table`` element
* The ``table`` is the sole element within the ``figure``
* Both ``figcaption`` and ``caption`` elements are in use

**Example of a potential warning message:**
> **Warning:** Element ``figure`` contains a ``figcaption`` element and one ``table`` element containing a ``caption`` element. Consider [omitting the ``caption`` element](https://html.spec.whatwg.org/multipage/tables.html#the-caption-element:the-table-element-5).

**Affected code sample:**
```html
<!DOCTYPE html>
<html lang="">
<head>
<title>Test</title>
</head>
<body>

<figure>
	<figcaption>Figure caption</figcaption>
	<table>
		<caption>Table caption</caption>
	</table>
</figure>

</body>
</html>
```

**Unaffected code sample (``figure`` contains more than just one ``table`` element):**
```html
<!DOCTYPE html>
<html lang="">
<head>
<title>Test</title>
</head>
<body>

<figure>
	<figcaption>Figure caption</figcaption>
	<table>
		<caption>Table caption</caption>
	</table>
	<p>Other content</p>
</figure>

</body>
</html>
```

**Comment by sideshowbarker** [comment]
Agreed this would be useful — and in general any statement in the spec that uses the word *should* is actually a normative statement. So any *should* statement in the spec that states document-conformance requirements (_aka_ authoring-conformance requirements) is arguably something the checker needs to be reporting a warning for — in order to be in conformance with the spec requirements.
